packages/twenty-front/src/modules/object-record/record-table/components/RecordTable.tsx

```diff
@@ -82,7 +83,11 @@ export const RecordTable = ({
         recordTableId={recordTableId}
         viewBarId={viewBarId}
       >
-        <RecordTableBodyEffect />
+        <RecordGroupContextProvider
```

Comment:
This pattern is never used in the codebase (and shouldn't if that's the case), we shouldn't abstract a .map in a component that will loop on its children. The preferred way is to be explicit about the fact that a component is displaying multiple items.

For exemple here it could be : 
- `RecordTableGroupBodyEffects`
- `RecordTableBodyEffect`

And we could use conditional rendering between the two at this level.

Another problem here is that the `ContextProvider` suffix should be used only to abstract some logic that will be put into the Context + the Context.Provider wrapping.

Here it hides a conditional rendering + a .map which should be made explicit at that level of the tree.

Also I'm not sure that we should put a RecordGroupContext if there's no actual record group, that would not be explicit and meaningful, and could confuse further feature development.

Thus having the tree splitting in two here, or even higher `RecordTableWithoutGroup` and `RecordTableWithGroups` (maybe too high but worth considering) ?

Also it's obvious after reading this PR that we should use a recoil state to know explicitly if we're in the case of having record groups or not, instead of encoding it in different places.

Trying to mix the no group case into the multiple group case with magic ids and ifs will create technical debt in the long run IMHO.

I'm not sure about how it will look like exactly though, but you get the idea here.